core/src/main/java/org/radargun/traits/Lifecycle.java

```diff
@@ -21,4 +21,11 @@
     * @return True if the service was started but not stopped.
     */
    boolean isRunning();
+
+   default void shouldWait(boolean wait) {
```

Comment:
Listening :)

Reply:
Just the empty implementation here mostly. It's related to making the `WaitLifecycle` class

Reply:
How about make `Lifecycle` an abstract class and remove `WaitLifecycle`?

Reply:
If that will work with the current class hierarchy of the `Lifecycle` implementations, then I would prefer that.

Reply:
We can't because `public class EHCacheService implements Lifecycle, Clustered {`
I don't see a different way to achieve that. Do you have something in mind ?

Reply:
We can't because `class JGroupsService extends ReceiverAdapter implements Lifecycle, Clustered, BasicOperations.Cache` do you have any other idea ?

Reply:
OK, so the other question is whether all of the current `Lifecycle` implementations have a good implementation of `shouldWait(boolean wait)` or `shouldWait()`? This would mean not having the `WaitLifecycle` class.

Reply:
The problem is that `Lifecycle` cannot be an abstract class and it should have the `boolean wait` state. Removing `WaitLifecycle` and making `Lifecycle` abstract will breaks `JGroupsService`. By default only class that extends `WaitLifecycle` will be affected by this behavior and the other classes will remains as it was before.